stockflux-securities-master/src/components/InputForm.js

```diff
@@ -27,47 +37,60 @@ const InputForm = ({ match }) => {
 
   useEffect(() => {
     if (match.params.securityId) {
-      setIsLoading(true);
+      setContextOrState(stateEnum.loading);
       getSecurity(match.params.securityId)
         .then(security => {
-          setIsLoading(false);
-          setErrorMessage(null);
+          setContextOrState(null);
           setSecurityState(security);
         })
         .catch(() => {
-          setIsLoading(false);
-          setErrorMessage("Error, cannot get security");
+          setContextOrState(stateEnum.error);
+          setMessages(["Error, cannot get security"]);
         });
     }
   }, []);
 
   const submitForm = event => {
     event.preventDefault();
-    setInProgress(true);
-
+    setContextOrState(stateEnum.sending);
     const securityObject = {
-      name,
       exchange,
       symbol,
+      name,
       visible,
       enabled
     };
 
-    // call service
-
-    setInProgress(false);
-    console.log(securityObject);
+    postSecurity(securityObject)
+      .then(() => {
+        setMessages(["Security was successfully created"]);
+        setContextOrState(stateEnum.success);
+        setSecurityState({
+          exchange: "",
+          symbol: "",
+          name: "",
+          visible: false,
+          enabled: false
+        });
+        setContextOrState(stateEnum.success);
```

Comment:
You've already called this once on line 66.

Reply:
Yeah, I need to be more careful to spot these obvious mistakes myself before I commit. Changed now.